packages/context/src/binding.ts

```diff
@@ -251,6 +306,36 @@ export class Binding {
     return this;
   }
 
+  /**
+   * Set options for the binding
+   * @param options Options object or a promise
+   */
+  withOptions(options: BindingOptions): this {
+    if (isPromise(options)) {
```

Comment:
Please move the code handling the complexity of merging `ValueOrPromise<Object>` with another `ValueOrPromise<Object>` into a helper function and simplify `withOptions` to stay at higher level.

```ts
withOptions(options: BindingOptions): this {
  this.options = mergeObjectOrPromise(this.options, options);
  return this;
}
```